### DISCUSSION TITLE
ALP compression for Floating point numbers

### DISCUSSION THREAD

**Discussion by MacNale** [ClickHouse/ClickHouse#60393] [Ideas]
Came across a youtube video discussing ALP compression. It looks ALP works well for Floating point numbers compression in vectors and faster.

<img width="822" alt="image" src="https://github.com/ClickHouse/ClickHouse/assets/14235758/510c68fa-448f-4d81-b4ec-1a554da6ddcf">

<img width="894" alt="image" src="https://github.com/ClickHouse/ClickHouse/assets/14235758/856f5cb9-ce19-4e84-99bd-4a3a9804c5d9">

Sources - 
https://www.youtube.com/watch?v=GKG5b04o5Yc
https://ir.cwi.nl/pub/33334

 1) Does clickhouse support ALP?
 2) What is best way to encode/compress Float64 today in clickhouse?

**Comment by mlex** [comment]
Hi!

> Does clickhouse support ALP?

No, not yet. The list of currently supported column compression codecs is in https://clickhouse.com/docs/en/sql-reference/statements/create/table#specialized-codecs.

Would you like to open a issue / feature request for that?

> What is best way to encode/compress Float64 today in clickhouse?

There is a [blog post](https://clickhouse.com/blog/optimize-clickhouse-codecs-compression-schema) where we analyzed compression codecs on an example dataset (the analysis for float columns it at the very end of the blog post). For float datatypes potential options are Gorilla, FPC or just plain zstd. Which one performs better depends a lot on the shape of the data.

**Reply by MacNale** [reply]
Thank you. Opened an issue - #60533
